src/cmd/oui/mod.rs

```diff
@@ -0,0 +1,85 @@
+use crate::{
+    cmd::{print_footer, print_json, status_json, status_str, Opts, OutputFormat},
+    keypair::PublicKey,
+    result::{anyhow, Result},
+    traits::b64::B64,
+};
+use helium_api::pending_transactions::PendingTxnStatus;
+use helium_proto::*;
+use serde_json::json;
+use std::convert::TryInto;
+use structopt::StructOpt;
+
+mod create;
+use create::*;
+mod update;
+use update::*;
+mod submit;
+use submit::*;
+
+/// Create or update an OUI
+#[derive(Debug, StructOpt)]
+pub enum Cmd {
+    Create(Create),
+    Submit(Submit),
+    Update(Update),
+}
+
+impl Cmd {
+    pub async fn run(&self, opts: Opts) -> Result {
+        match self {
+            Cmd::Create(cmd) => cmd.run(opts).await,
+            Cmd::Submit(cmd) => cmd.run(opts).await,
+            Cmd::Update(cmd) => cmd.run(opts).await,
+        }
+    }
+}
+
+fn map_addresses<F, R>(addresses: Vec<impl TryInto<PublicKey>>, f: F) -> Result<Vec<R>>
+where
+    F: Fn(PublicKey) -> R,
+{
+    let results: Result<Vec<R>> = addresses
+        .into_iter()
+        .map(|v| match v.try_into() {
+            Ok(public_key) => Ok(f(public_key)),
+            Err(_err) => Err(anyhow!("failed to convert to public key")),
+        })
+        .collect();
+    results
+}
+
+fn print_txn(
+    txn: &BlockchainTxnOuiV1,
+    envelope: &BlockchainTxn,
+    status: &Option<PendingTxnStatus>,
+    format: OutputFormat,
+) -> Result {
+    match format {
+        OutputFormat::Table => {
+            ptable!(
+                ["Key", "Value"],
+                ["Requested OUI", txn.oui],
```

Comment:
It's not really the requested OUI right? It's the last known OUI? Hard to put good words to it, I know

Reply:
I'm a little baffled now, because the transaction works with this:
```
ouis::last(&client).await?.oui + 1
```

So now when I display the OUI, no more incrementing. And this clears on testnet...

I'm really confused. Before, I thought the create transaction used the last OUI (no increment). So either I misunderstood/remembered what was happening before or the testnet has a change to blockchain core? I'm truly baffled.

Reply:
I renamed this to "Last OUI"; that way, we don't have to lie and it it hopefully explains itself to the user